pkg/generators/cluster.go

```diff
@@ -65,7 +65,10 @@ func (g *ClusterGenerator) GenerateParams(
 	// For each matching cluster secret
 	res := make([]map[string]string, len(clusterSecretList.Items))
 	for i, cluster := range clusterSecretList.Items {
-		params := make(map[string]string, len(cluster.ObjectMeta.Annotations)+len(cluster.ObjectMeta.Labels)+2)
+		params := make(map[string]string, len(appSetGenerator.Clusters.Values)+len(cluster.ObjectMeta.Annotations)+len(cluster.ObjectMeta.Labels)+2)
+		for key, value := range appSetGenerator.Clusters.Values {
+			params[key] = value
```

Comment:
any thoughts on prefixing w/ something like
 ```go
        params[fmt.Sprintf("metadata.values.%s", key)] = value
```
 here instead, and avoiding stomping on  other things in params[]?

Reply:
+1 for allowing values to be passed through params, many use cases there for templating those params in helm as control variables in a values.yaml

Reply:
> any thoughts on prefixing w/ something like
> 
> ```go
>        params[fmt.Sprintf("metadata.values.%s", key)] = value
> ```
> 
> here instead, and avoiding stomping on other things in params[]?

I have no objections to that, though I think we should drop the `metadata.` prefix. Could cause confusion with `metadata.labels` and `metadata.annotations` coming from one place in the AppSet and `metadata.value` from another.

I propose:

 ```go
        params[fmt.Sprintf("values.%s", key)] = value
 ```